nmigen/vendor/intel.py

```diff
@@ -371,3 +371,36 @@ def get_diff_input_output(self, pin, p_port, n_port, attrs, invert):
             i_oe=self._get_oereg(m, pin),
         )
         return m
+
+    def get_ff_sync(self, ff_sync):
+        assert ff_sync._stages >= 2, "the synchronizer must be at least 2 stages long"
```

Comment:
This isn't the place to add this assert (maybe FFSynchronizer itself is, but not in this PR).